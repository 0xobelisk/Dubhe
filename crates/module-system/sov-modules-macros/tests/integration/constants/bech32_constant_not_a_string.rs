use sov_bank::TokenId;
use sov_modules_api::macros::config_bech32;

const TEST_ARRAY_OF_U8: TokenId = config_bech32!("TEST_ARRAY_OF_U8", TokenId);

fn main() {
    assert_eq!(TEST_ARRAY_OF_U8, TokenId::from([0; 32]));
}
