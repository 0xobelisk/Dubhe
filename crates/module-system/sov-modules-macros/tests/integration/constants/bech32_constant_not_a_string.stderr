error: The constant named `TEST_ARRAY_OF_U8` inside `$WORKSPACE/constants.testing.toml` should be a bech32 string, but it's not a string in the first place: `[11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11,
       11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11]`
 --> tests/integration/constants/bech32_constant_not_a_string.rs:4:50
  |
4 | const TEST_ARRAY_OF_U8: TokenId = config_bech32!("TEST_ARRAY_OF_U8", TokenId);
  |                                                  ^^^^^^^^^^^^^^^^^^
